airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/message/DestinationMessageQueueWriterTest.kt

```diff
@@ -5,40 +5,25 @@
 package io.airbyte.cdk.message
 
 import com.fasterxml.jackson.databind.node.JsonNodeFactory
-import io.airbyte.cdk.command.DestinationCatalog
 import io.airbyte.cdk.command.DestinationStream
 import io.airbyte.cdk.command.MockCatalogFactory.Companion.stream1
 import io.airbyte.cdk.command.MockCatalogFactory.Companion.stream2
 import io.airbyte.cdk.data.NullValue
 import io.airbyte.cdk.state.CheckpointManager
-import io.micronaut.context.annotation.Prototype
+import io.micronaut.context.annotation.Requires
 import io.micronaut.test.extensions.junit5.annotation.MicronautTest
 import jakarta.inject.Inject
-import jakarta.inject.Named
+import javax.inject.Singleton
 import kotlinx.coroutines.test.runTest
 import org.junit.jupiter.api.Assertions
 import org.junit.jupiter.api.Test
 
-@MicronautTest(environments = ["MockStreamsManager"])
+@MicronautTest(environments = ["MockStreamsManager"], rebuildContext = true)
 class DestinationMessageQueueWriterTest {
-    @Inject lateinit var queueWriterFactory: TestDestinationMessageQueueWriterFactory
-
-    @Prototype
-    class TestDestinationMessageQueueWriterFactory(
-        @Named("mockCatalog") private val catalog: DestinationCatalog,
-        val messageQueue: MockMessageQueue,
-        val streamsManager: MockStreamsManager,
-        val checkpointManager: MockCheckpointManager
-    ) {
-        fun make(): DestinationMessageQueueWriter {
-            return DestinationMessageQueueWriter(
-                catalog,
-                messageQueue,
-                streamsManager,
-                checkpointManager
-            )
-        }
-    }
+    @Inject lateinit var writer: DestinationMessageQueueWriter
+    @Inject lateinit var messageQueue: MockMessageQueue
+    @Inject lateinit var streamsManager: MockStreamsManager
```

Comment:
minor question: is it better practice to declare using the concrete Mock class, or the interface `StreamsManager`? (declaring the mock makes it more obvious that this is a mock, but declaring as the interface makes it easier to notice conflicting beans

Reply:
I guess this could also be `lateinit var mockWhatever: Whatever`, which kind of gets the best of both worlds?

Reply:
My main motivation for declaring it as the mock is because my intent is usually to use methods I added specifically for validation. Otherwise I have to cast.

However, you're right that using the interface helps make it clear what's getting injected. The little bean should point to the mock.